cosmos/operators/aws_ecs.py

```diff
@@ -68,25 +71,41 @@ def __init__(
                 "aws_conn_id": aws_conn_id,
                 "task_definition": task_definition,
                 "cluster": cluster,
-                "container_name": container_name,
                 "overrides": None,
             }
         )
-        super().__init__(**kwargs)
+
+        if provider_version >= "9.3.0":
+            kwargs["container_name"] = container_name
+
         # In PR #1474, we refactored cosmos.operators.base.AbstractDbtBase to remove its inheritance from BaseOperator
         # and eliminated the super().__init__() call. This change was made to resolve conflicts in parent class
         # initializations while adding support for ExecutionMode.AIRFLOW_ASYNC. Operators under this mode inherit
         # Airflow provider operators that enable deferrable SQL query execution. Since super().__init__() was removed
         # from AbstractDbtBase and different parent classes require distinct initialization arguments, we explicitly
         # initialize them (including the BaseOperator) here by segregating the required arguments for each parent class.
-        base_operator_args = set(inspect.signature(EcsRunTaskOperator.__init__).parameters.keys())
+        default_args = kwargs.get("default_args", {})
+        operator_kwargs = {**kwargs}
+
+        operator_args: set[str] = set()
+        for clazz in EcsRunTaskOperator.__mro__:
+            operator_args.update(inspect.signature(clazz.__init__).parameters.keys())
+            if clazz == BaseOperator:
+                break
+
         base_kwargs = {}
-        for arg_key, arg_value in kwargs.items():
-            if arg_key in base_operator_args:
-                base_kwargs[arg_key] = arg_value
-        base_kwargs["task_id"] = kwargs["task_id"]
-        base_kwargs["aws_conn_id"] = aws_conn_id
-        EcsRunTaskOperator.__init__(self, **base_kwargs)
+        for arg in {*inspect.signature(AbstractDbtBase.__init__).parameters.keys()}:
+            try:
+                base_kwargs[arg] = kwargs[arg]
+                if arg not in operator_args:
+                    operator_kwargs.pop(arg)
+            except KeyError:
+                try:
+                    base_kwargs[arg] = default_args[arg]
+                except KeyError:
+                    pass
```

Comment:
```suggestion
            if arg in kwargs:
                base_kwargs[arg] = kwargs[arg]
                if arg not in operator_args:
                    operator_kwargs.pop(arg, None)
            elif arg in default_args:
                base_kwargs[arg] = default_args[arg]
```

This should yield the same result, but without the need for nested try-except KeyError handling. WDYT? If it looks fine, then similar suggestion for other places in the PR.

Reply:
I don't feel very strongly about it, but I've heard the try except is preferred because its a single lookup to the dictionary instead of checking if the key exists and then getting the value for it.